src/index.ts

```diff
@@ -148,23 +148,33 @@ export const createRouter = <
     return route ? JSON.parse(route) : route;
   };
 
+  type CompareHrefUrl = { href: string; currentLocation: string };
+  const defaultCompareUrls = ({ href, currentLocation }: CompareHrefUrl) =>
+    href === currentLocation;
+
   // Kudos to https://github.com/remix-run/react-router/pull/7998
   const useLink = ({
     href,
     replace = false,
     target,
+    compareUrls = defaultCompareUrls,
   }: {
     href: string;
     replace?: boolean | undefined;
     target?: React.HTMLAttributeAnchorTarget | undefined;
+    compareUrls?: (params: CompareHrefUrl) => boolean;
   }) => {
     const active = useSubscription(
       React.useMemo(
         () => ({
-          getCurrentValue: () => href === getCurrentLocation().url,
+          getCurrentValue: () =>
+            compareUrls({
+              href,
+              currentLocation: getCurrentLocation().url,
```

Comment:
Instead of passing `currentLocation.url`, passing the whole `Location` object could be easier to use

Reply:
`currentLocation` -> `location` for coherence (`currentLocation` is only used internaly - we expose a `getLocation` fn)